cmd/updateIntegrationArtifactConfiguration.go

```diff
@@ -0,0 +1,111 @@
+		Command: &command.Command{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func updateIntegrationArtifactConfiguration(config updateIntegrationArtifactConfigurationOptions, telemetryData *telemetry.CustomData) {
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
+	httpClient := &piperhttp.Client{}
+	// For HTTP calls import  piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	// and use a  &piperhttp.Client{} in a custom system
+	// Example: step checkmarxExecuteScan.go
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runUpdateIntegrationArtifactConfiguration(&config, telemetryData, httpClient)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runUpdateIntegrationArtifactConfiguration(config *updateIntegrationArtifactConfigurationOptions, telemetryData *telemetry.CustomData, httpClient piperhttp.Sender) error {
+	clientOptions := piperhttp.ClientOptions{}
+
+	configUpdateURL := fmt.Sprintf("%s/api/v1/IntegrationDesigntimeArtifacts(Id='%s',Version='%s')/$links/Configurations('%s')", config.Host, config.IntegrationFlowID, config.IntegrationFlowVersion, config.ParameterKey)
+	tokenParameters := cpi.TokenParameters{TokenURL: config.OAuthTokenProviderURL, Username: config.Username, Password: config.Password, Client: httpClient}
+	token, err := cpi.CommonUtils.GetBearerToken(tokenParameters)
+	if err != nil {
+		return errors.Wrap(err, "failed to fetch Bearer Token")
+	}
+	clientOptions.Token = fmt.Sprintf("Bearer %s", token)
+	httpClient.SetOptions(clientOptions)
+	httpMethod := "PUT"
+	header := make(http.Header)
+	header.Add("Content-Type", "application/json")
+	header.Add("Accept", "application/json")
+	jsonObj := gabs.New()
+	jsonObj.Set(config.ParameterValue, "ParameterValue")
+	jsonBody, jsonErr := json.Marshal(jsonObj)
+
+	if jsonErr != nil {
+		return errors.Wrap(jsonErr, "input json body is invalid")
+	}
+	configUpdateResp, httpErr := httpClient.SendRequest(httpMethod, configUpdateURL, bytes.NewBuffer(jsonBody), header, nil)
+	if httpErr != nil {
+		return errors.Wrapf(httpErr, "HTTP %q request to %q failed with error", httpMethod, configUpdateURL)
+	}
+
+	if configUpdateResp != nil && configUpdateResp.Body != nil {
+		defer configUpdateResp.Body.Close()
+	}
+
+	if configUpdateResp == nil {
+		return errors.Errorf("did not retrieve a HTTP response")
+	}
+
+	if configUpdateResp.StatusCode == http.StatusAccepted {
+		log.Entry().
+			WithField("IntegrationFlowID", config.IntegrationFlowID).
+			Info("successfully updated the integration flow configuration parameter")
+		return nil
+	}
+	response, readErr := ioutil.ReadAll(configUpdateResp.Body)
+
+	if readErr != nil {
```

Comment:
I'm complaining on a very high level now, sorry ...

I would like to suggest to mention the response code in case of a body-read error. Otherwise we don't know which http-code we received and we have only the information that the body was not readable. That would be a poor situation during a troubleshooting session

Reply:
Would you like to include the status code in the line below in order to not only know that we cannot read the body, but also the status code. Might be useful during trouble shooting.